Tasks/AzureVmssDeployment/Resources/customScriptInvoker.sh

```diff
@@ -1,5 +1,12 @@
 #!/bin/bash
-tar -xzC . -f $1
+if [ -n "$1" ]; then
```

Comment:
what does this mean? wont the 2nd param become the 1s one in that case. Should you rather check $3

Reply:
this is checking if there is a targz which needs to be extracted. $1 is targz name. $2 and $3 will be used to run command irrespective of whether extract was done or whether loose files were directly available.